src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/volume/v4_9/CloudStackVolumePluginTest.java

```diff
@@ -72,22 +71,148 @@
+        this.cloudStackUser = CloudstackTestUtils.CLOUD_STACK_USER;
+
         this.client = Mockito.mock(CloudStackHttpClient.class);
-        this.plugin = new CloudStackVolumePlugin(cloudStackConfFilePath);
+        this.plugin = Mockito.spy(new CloudStackVolumePlugin(cloudStackConfFilePath));
         this.plugin.setClient(this.client);
-        this.cloudUser = new CloudStackUser(FAKE_USER_ID, FAKE_NAME, FAKE_TOKEN_VALUE, FAKE_DOMAIN, FAKE_COOKIE_HEADER);
+
+        this.testUtils.mockReadOrdersFromDataBase();
+        CloudstackTestUtils.ignoringCloudStackUrl();
+    }
+
+    // test case: When calling the deleteInstance method with secondary methods mocked,
+    // it must verify if the doDeleteInstance is called with the right parameters;
+    // this includes the checking of the Cloudstack request.
+    @Test
+    public void testDeleteInstanceSuccessfully() throws FogbowException {
+        // set up
+        String instanceId = "instanceId";
+        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
+        Mockito.when(volumeOrder.getInstanceId()).thenReturn(instanceId);
+
+        Mockito.doNothing().when(this.plugin).doDeleteInstance(Mockito.any(), Mockito.eq(this.cloudStackUser));
+
+        DeleteVolumeRequest request = new DeleteVolumeRequest.Builder()
+                .id(instanceId)
+                .build(this.cloudStackUrl);
+
+        // exercise
+        this.plugin.deleteInstance(volumeOrder, this.cloudStackUser);
+
+        // verify
+        RequestMatcher<DeleteVolumeRequest> matcher = new RequestMatcher.DeleteVolume(request);
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE))
+                .doDeleteInstance(Mockito.argThat(matcher), Mockito.eq(this.cloudStackUser));
+    }
+
+    // test case: When calling the deleteInstance method and occurs an FogbowException,
+    // it must verify if It returns a FogbowException.
+    @Test(expected = FogbowException.class)
+    public void testDeleteInstanceFail() throws FogbowException {
+        // set up
+        String instanceId = "instanceId";
+        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
+        Mockito.when(volumeOrder.getInstanceId()).thenReturn(instanceId);
+
+        Mockito.doThrow(new FogbowException())
+                .when(this.plugin).doDeleteInstance(Mockito.any(), Mockito.eq(this.cloudStackUser));
+
+        // exercise
+        this.plugin.deleteInstance(volumeOrder, this.cloudStackUser);
+    }
+
+    // test case: When calling the doDeleteInstance method with secondary methods mocked,
+    // it must verify if It reachs the right method without problems.
+    @Test
+    public void testDoDeleteInstanceSuccessfully() throws FogbowException, HttpResponseException {
+        // set up
+        DeleteVolumeRequest request = new DeleteVolumeRequest.Builder().build("");
+
+        String responseStr = "anything";
```

Comment:
Use the **_ANY_VALUE_** constant of the **TestUtils** class... Do this for other occurrences.

Reply:
Done. 
Note: I've done the changes in the fourth PR in order to avoid conflicts.